.pre-commit-config.yaml

```diff
@@ -1,28 +1,43 @@
 default_language_version:
-    python: python3.10
+  python: python3.10
+
+fail_fast: true
 
 repos:
--   repo: https://github.com/psf/black
-    rev: 22.10.0
-    hooks:
-    -   id:   black
-        exclude: migrations
--   repo: https://github.com/pycqa/isort
+  - repo: https://github.com/pycqa/isort
     rev: 5.10.1
     hooks:
       - id: isort
         args: ["--profile", "black"]
         name: isort (python)
 
--   repo: https://github.com/PyCQA/autoflake
-    rev: v1.7.7
+  - repo: https://github.com/psf/black
+    rev: 22.10.0
     hooks:
-    - id: autoflake
-      args: ['--in-place', '--remove-all-unused-imports', '--expand-star-imports', '--remove-duplicate-keys', '--remove-unused-variables']
+      - id: black
 
-# uncomment once the repo complies with flake8
-#-   repo: https://github.com/pycqa/flake8
-#    rev: 5.0.4
-#    hooks:
-#    -   id: flake8
-#        args: ["--ignore=E501, W503"]
\ No newline at end of file
+  - repo: https://github.com/pycqa/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+        args:
+          ["--ignore=E402, E501, W503, E712, E713, E722"]
+          # E402 = module level import not at top of file
+          # E501 = line too long
+          # W503 = line break occurred before a binary operator
+          # E712 = comparison to False should be 'if cond is False:' or 'if not cond:'
```

Comment:
We'll eventually want to remove this, but for now, it's fine.

Reply:
Actually the rest after this

Reply:
Yes, they are mostly inconsequential and mainly there to get the important parts of flake8 into the pre-commit workflow